tests/credentials_provider_cognito_tests.c

```diff
@@ -816,3 +824,135 @@ static int s_credentials_provider_cognito_success_dynamic_token_pairs_fn(struct
 AWS_TEST_CASE(
     credentials_provider_cognito_success_dynamic_token_pairs,
     s_credentials_provider_cognito_success_dynamic_token_pairs_fn);
+
+static int s_credentials_provider_cognito_proxy_routing_enabled_test(struct aws_allocator *allocator, void *ctx) {
+    (void)ctx;
+
+    s_aws_cognito_tester_init(allocator);
+
+    struct aws_string *proxy_env = aws_string_new_from_c_str(allocator, "HTTP_PROXY");
+    struct aws_string *proxy_val = aws_string_new_from_c_str(allocator, "http://fake-proxy:9999");
+    aws_set_environment_value(proxy_env, proxy_val);
+    aws_string_destroy(proxy_env);
+    aws_string_destroy(proxy_val);
+
+    struct proxy_env_var_settings proxy_config = {
+        .env_var_type = AWS_HPEV_ENABLE,
+    };
+
+    s_tester.proxy_config = &proxy_config;
+
+    struct aws_credentials_provider_cognito_options options = {
+        .bootstrap = s_tester.bootstrap,
+        .function_table = &s_mock_function_table,
+        .endpoint = aws_byte_cursor_from_c_str("somewhere.amazonaws.com"),
+        .identity = aws_byte_cursor_from_c_str("someone"),
+        .tls_ctx = s_tester.ctx,
+        .proxy_ev_settings = &proxy_config,
+    };
```

Comment:
the test just go through the mock that don't actually read the environment variables. The goal is to check everything in `s_tester.proxy_config` passed down to `s_aws_http_connection_manager_new_mock` correctly. So, I'd just set the other field and remove the env variable in the test.
And, I don't think we need three different tests for this one thing